### DISCUSSION TITLE
issues upgrading php7.2 to 8.1

### DISCUSSION THREAD

**Discussion by iainmsi** [rectorphp/rector#7756] [Q&A] [answered]
Hi I tried running rector to upgrade the php 7.2 to 8.1 on old application
and I've got a few errors and I don't know what to do about it and was hoping someone could help? please
my rector.php
<?php

declare(strict_types=1);

use Rector\CodeQuality\Rector\Class_\InlineConstructorDefaultToPropertyRector;
use Rector\Config\RectorConfig;
use Rector\Set\ValueObject\LevelSetList;

return static function (RectorConfig $rectorConfig): void {
        $rectorConfig->import(LevelSetList::UP_TO_PHP_81);

};

errors i'm seeing 
[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/app/library/ecl/dictionary.php on line 88

Deprecated: Return type of Ecl_Dictionary::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/app/library/ecl/dictionary.php on line 94

Deprecated: Return type of Ecl_Dictionary::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/app/library/ecl/dictionary.php on line 100

Deprecated: Return type of Ecl_Dictionary::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /var/www/app/library/ecl/dictionary.php on line 106

Notice: spl_autoload_register(): Argument #2 ($do_throw) has been ignored, spl_autoload_register() will always throw in /var/www/app/library/ecl/mvc/model.php on line 44

Fatal error: Uncaught Error: Call to undefined function ldap_connect() in /var/www/app/library/ecl/ldap.php:102 Stack trace: #0 /var/www/app/models/authenticator.php(74): Ecl_Ldap->connect() #1 /var/www/app/controllers/signin.php(30): Authenticator->authenticateLdap() #2 /var/www/app/library/ecl/mvc/controller.php(84): Controller_Signin->actionCheck() #3 /var/www/app/library/ecl/mvc/router.php(123): Ecl_Mvc_Controller->action() #4 /var/www/app/library/ecl/mvc/router.php(578): Ecl_Mvc_Router->action() #5 /var/www/app/library/ecl/mvc/router.php(286): Ecl_Mvc_Router->_dispatchAction() #6 /var/www/index.php(101): Ecl_Mvc_Router->dispatch() #7 {main} thrown in /var/www/app/library/ecl/ldap.php on line 102

**Comment by samsonasik** [comment, marked as answer]
You can add `ReturnTypeWillChangeRector` to config:

```php
<?php

declare(strict_types=1);

use Rector\Config\RectorConfig;
use Rector\Transform\Rector\ClassMethod\ReturnTypeWillChangeRector;
use Rector\Transform\ValueObject\ClassMethodReference;

return static function (RectorConfig $rectorConfig): void {
    // ... other config

    $rectorConfig->ruleWithConfiguration(ReturnTypeWillChangeRector::class, [
        new ClassMethodReference('ArrayAccess', 'offsetSet')
    ]);
};
```

See https://getrector.com/demo/84796f9c-00bc-4c2d-aa3f-9dc1c9fbc580

**Comment by iainmsi** [comment]
thanks for your reply i'm still seeing these errors

Notice: spl_autoload_register(): Argument #2 ($do_throw) has been ignored,
spl_autoload_register() will always throw in /var/www/app/library/ecl.php
on line 223

Deprecated: Return type of Ecl_Dictionary::offsetExists($offset) should
either be compatible with ArrayAccess::offsetExists(mixed $offset): bool,
or the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in /var/www/app/library/ecl/dictionary.php on line 88

Deprecated: Return type of Ecl_Dictionary::offsetUnset($offset) should
either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or
the #[\ReturnTypeWillChange] attribute should be used to temporarily
suppress the notice in /var/www/app/library/ecl/dictionary.php on line 108

Deprecated: Creation of dynamic property Ecl_Mvc_Router::$_time_started is
deprecated in /var/www/app/library/ecl/mvc/router.php on line 63

Notice: spl_autoload_register(): Argument #2 ($do_throw) has been ignored,
spl_autoload_register() will always throw in
/var/www/app/library/ecl/mvc/model.php on line 44

Deprecated: strtolower(): Passing null to parameter #1 ($string) of type
string is deprecated in /var/www/app/views/signin/index.phtml on line 31

**Reply by samsonasik** [reply]
You can add more `class->method` configurable to `ReturnTypeWillChangeRector` as above :) 

For deprecated dynamic property, you can apply `CompleteDynamicPropertiesRector`, see https://getrector.com/demo/ff62f32d-94ee-4c6f-a417-c991c0e4c48f
